### DISCUSSION TITLE
How cover the keyboard in custom editor

### DISCUSSION THREAD

**Discussion by ITfanheng** [eclipse-theia/theia#11550] [Q&A]
Hi everyone!

I create a widget which extends `BaseWidget`.

<img width="752" alt="截屏2022-08-10 上午10 03 03" src="https://user-images.githubusercontent.com/75559678/183794197-328ad70c-3b51-4c12-8a0f-532b6ea5f818.png">

And the keyboard of this ui framework was covered by theia.

Is there any way to use theia keyboard or cover theia keyboard.

Thanks in advance!

**Comment by msujew** [comment]
Hey @ITfanheng,

what do you mean with "covering" exactly? Maybe that the keyboard shortcuts of your widget are shadowed by Theia's shortcuts?

**Reply by ITfanheng** [reply]
yes, you are right

**Comment by maheshrajrp** [comment]
Hi,
You could use `unregisterKeybinding` or based on the keybinding you could also use `when` clause.
 
```
@injectable()
export class CustomKeybindingContribution implements KeybindingContribution {
    /**
     * Registers keybindings.
     * @param keybindings the keybinding registry.
     */

    @inject(CommandRegistry)
    commandRegistry: CommandRegistry;

    registerKeybindings(keybindings: KeybindingRegistry) {
        keybindings.unregisterKeybinding(this.commandRegistry.getCommand(CommonCommands.SAVE.id));// To remove ctrlcmd+s shortcut
    }
}
```
